### ISSUE TITLE
UNHANDLED REJECTION Invalid options object. React Refresh Plugin...

### ISSUE THREAD

**Issue by rchrdnsh** [gatsbyjs/gatsby#24078]
Updated to the latest version of gatsby and am getting this error trying to run develop mode:

```bash
RN:RYKR rchrdnsh$ gatsby develop
success open and validate gatsby-configs - 0.059s
success load plugins - 2.807s
success onPreInit - 0.012s
success initialize cache - 0.010s
success copy gatsby files - 0.043s
success onPreBootstrap - 0.009s
success createSchemaCustomization - 0.007s
success source and transform nodes - 3.956s
success building schema - 0.280s
success createPages - 0.060s
success createPagesStatefully - 0.092s
success onPreExtractQueries - 0.002s
success update schema - 0.023s
success extract queries from components - 0.603s
success write out requires - 0.041s
success write out redirect data - 0.004s
success Build manifest and related icons - 0.098s
success onPostBootstrap - 0.107s
⠀
info bootstrap finished - 11.514s
⠀
success run queries - 6.769s - 38/38 5.61/s
⠋ Building development bundle
[============================]   17.593 s 118/118 100% Generating image thumbnails

 ERROR 

UNHANDLED REJECTION Invalid options object. React Refresh Plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'disableRefreshCheck'. These properties are valid:
   object { exclude?, forceEnable?, include?, overlay?, useLegacyWDSSockets? }



  ValidationError: Invalid options object. React Refresh Plugin has been initialized using an options object that does not match the API schema.
   - options has an unknown property 'disableRefreshCheck'. These properties are valid:
     object { exclude?, forceEnable?, include?, overlay?, useLegacyWDSSockets? }
  
  - validate.js:88 validate
    [RYKR]/[schema-utils]/dist/validate.js:88:11
  
  - index.js:18 new ReactRefreshPlugin
    [RYKR]/[@pmmmwh]/react-refresh-webpack-plugin/src/index.js:18:5
  
  - webpack-utils.js:499 Object.plugins.fastRefresh
    [RYKR]/[gatsby]/dist/utils/webpack-utils.js:499:31
  
  - webpack.config.js:218 getPlugins
    [RYKR]/[gatsby]/dist/utils/webpack.config.js:218:107
  
  - webpack.config.js:433 module.exports
    [RYKR]/[gatsby]/dist/utils/webpack.config.js:433:14
  
  - develop.js:162 async startServer
    [RYKR]/[gatsby]/dist/commands/develop.js:162:21
  
  - develop.js:466 async module.exports
    [RYKR]/[gatsby]/dist/commands/develop.js:466:7
 

not finished Building development bundle - 0.186s
not finished Generating image thumbnails - 17.788s
```

no idea what's going on here.

Here is the Info report:

```bash
RN:RYKR rchrdnsh$ gatsby info

  System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 13.7.0 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.14.5 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 80.0.3987.163
    Edge: 80.0.361.111
    Firefox: 75.0
    Safari: 13.1
  npmPackages:
    gatsby: ^2.21.28 => 2.21.29
    gatsby-cli: ^2.12.19 => 2.12.20
    gatsby-image: ^2.4.3 => 2.4.3
    gatsby-plugin-layout: ^1.3.1 => 1.3.1
    gatsby-plugin-manifest: ^2.4.3 => 2.4.3
    gatsby-plugin-mdx: ^1.2.7 => 1.2.7
    gatsby-plugin-netlify-functions: 0.0.3 => 0.0.3
    gatsby-plugin-offline: ^2.2.2 => 2.2.10
    gatsby-plugin-preact: ^3.3.1 => 3.3.1
    gatsby-plugin-react-helmet: ^3.3.1 => 3.3.1
    gatsby-plugin-sharp: ^2.6.3 => 2.6.3
    gatsby-plugin-stripe: ^1.2.4 => 1.2.4
    gatsby-plugin-styled-components: ^3.3.1 => 3.3.1
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4
    gatsby-remark-images: ^3.3.3 => 3.3.3
    gatsby-source-filesystem: ^2.3.3 => 2.3.3
    gatsby-transformer-remark: ^2.8.8 => 2.8.8
    gatsby-transformer-sharp: ^2.5.2 => 2.5.2
    gatsby-transformer-yaml: ^2.4.1 => 2.4.1
  npmGlobalPackages:
    gatsby-cli: 2.12.19
```

**Comment by pieh** [comment]
Thanks for the report!

Appears that this bump https://github.com/gatsbyjs/gatsby/pull/23917 yesterday broke it ( https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/e0ddc1e3f1f2af3ab0637d40000791d7ee571751/CHANGELOG.md#030-10-may-2020 ). 

You should be able to revert back to previous version of gatsby while we get to fixing it or if you use yarn resolutions - you could add:
```
"resolutions": {
  "@pmmmwh/react-refresh-webpack-plugin": "0.2.0"
}
```

To force refresh react refresh plugin version that works with our current integration.

---

Note for whoever will work on the fix - to reproduce:
```
gatsby new i24078
cd i24078
yarn add gatsby@2.21.30
GATSBY_HOT_LOADER=fast-refresh yarn develop
```

**Comment by pieh** [comment]
Published `gatsby@2.21.31` which contains the fix - please upgrade and let us know if you run into any troubles. I did check locally and it worked then ... but you never know :)

**Comment by rchrdnsh** [comment]
All good again, thank you @pieh XD
